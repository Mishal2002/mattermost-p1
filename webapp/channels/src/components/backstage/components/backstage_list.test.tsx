// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';
import {FormattedMessage} from 'react-intl';

import BackstageList from './backstage_list';

describe('components/backstage/components/BackstageList', () => {
    test('Should have the browsing hint', () => {
        const wrapper = shallow(
            <BackstageList
                header={
                    <FormattedMessage
                        id='installed_incoming_webhooks.header'
                        defaultMessage='Installed Incoming Webhooks'
                    />
                }
                hintText={
                    <FormattedMessage
                        id='installed_incoming_webhooks.hint'
                        defaultMessage='Search by title of the webhook or by associated channel. Examples: "My Webhook Title", "town-square", or "Town Square".'
                    />
                }
                loading={false}
            >
                {() => {
                    return [[], false];
                }}
            </BackstageList>,
        );
        expect(wrapper.find(FormattedMessage).length === 2).toBe(true);
    });

    test("Shouldn't have the browsing hint", () => {
        const wrapper = shallow(
            <BackstageList
                header={
                    <FormattedMessage
                        id='installed_incoming_webhooks.header'
                        defaultMessage='Installed Incoming Webhooks'
                    />
                }
                loading={false}
            >
                {() => {
                    return [[], false];
                }}
            </BackstageList>,
        );
        expect(wrapper.find(FormattedMessage).length === 1).toBe(true);
    });

    test('Should match snapshot with browsing hint', () => {
        const wrapper = shallow(
            <BackstageList
                header={
                    <FormattedMessage
                        id='installed_incoming_webhooks.header'
                        defaultMessage='Installed Incoming Webhooks'
                    />
                }
                hintText={
                    <FormattedMessage
                        id='installed_incoming_webhooks.hint'
                        defaultMessage='Search by title of the webhook or by associated channel. Examples: "My Webhook Title", "town-square", or "Town Square".'
                    />
                }
                loading={false}
            >
                {() => {
                    return [[], false];
                }}
            </BackstageList>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('Should match snapshot without browsing hint', () => {
        const wrapper = shallow(
            <BackstageList
                header={
                    <FormattedMessage
                        id='installed_incoming_webhooks.header'
                        defaultMessage='Installed Incoming Webhooks'
                    />
                }
                loading={false}
            >
                {() => {
                    return [[], false];
                }}
            </BackstageList>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
