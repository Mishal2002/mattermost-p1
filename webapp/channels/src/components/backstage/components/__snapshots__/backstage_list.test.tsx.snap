// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/backstage/components/BackstageList Should match snapshot with browsing hint 1`] = `
<div
  className="backstage-content"
>
  <div
    className="backstage-header"
  >
    <h1>
      <MemoizedFormattedMessage
        defaultMessage="Installed Incoming Webhooks"
        id="installed_incoming_webhooks.header"
      />
    </h1>
  </div>
  <div
    className="backstage-filters"
  >
    <div
      className="backstage-filter__search"
    >
      <SearchIcon />
      <input
        className="form-control"
        id="searchInput"
        onChange={[Function]}
        placeholder="Search"
        style={
          Object {
            "flexGrow": 0,
            "flexShrink": 0,
          }
        }
        type="search"
        value=""
      />
    </div>
  </div>
  <span
    className="backstage-list__hint"
  >
    <MemoizedFormattedMessage
      defaultMessage="Search by title of the webhook or by associated channel. Examples: \\"My Webhook Title\\", \\"town-square\\", or \\"Town Square\\"."
      id="installed_incoming_webhooks.hint"
    />
  </span>
  <span
    className="backstage-list__help"
  />
  <div
    className="backstage-list"
  />
</div>
`;

exports[`components/backstage/components/BackstageList Should match snapshot without browsing hint 1`] = `
<div
  className="backstage-content"
>
  <div
    className="backstage-header"
  >
    <h1>
      <MemoizedFormattedMessage
        defaultMessage="Installed Incoming Webhooks"
        id="installed_incoming_webhooks.header"
      />
    </h1>
  </div>
  <div
    className="backstage-filters"
  >
    <div
      className="backstage-filter__search"
    >
      <SearchIcon />
      <input
        className="form-control"
        id="searchInput"
        onChange={[Function]}
        placeholder="Search"
        style={
          Object {
            "flexGrow": 0,
            "flexShrink": 0,
          }
        }
        type="search"
        value=""
      />
    </div>
  </div>
  <span
    className="backstage-list__hint"
  />
  <span
    className="backstage-list__help"
  />
  <div
    className="backstage-list"
  />
</div>
`;
